matchms/networking/SimilarityNetwork.py

```diff
@@ -0,0 +1,135 @@
+import networkx as nx
+import numpy
+from matchms import Scores
+from .networking_functions import get_top_hits
+
+
+class SimilarityNetwork:
+    """Create a spectal network from spectrum similarities.
+
+    For example
+
+    .. testcode::
+
+        import numpy as np
+        from matchms import Spectrum, calculate_scores
+        from matchms.similarity import ModifiedCosine
+        from matchms.networking import SimilarityNetwork
+
+        spectrum_1 = Spectrum(mz=np.array([100, 150, 200.]),
+                              intensities=np.array([0.7, 0.2, 0.1]),
+                              metadata={"precursor_mz": 100.0,
+                                        "testID": "one"})
+        spectrum_2 = Spectrum(mz=np.array([104.9, 140, 190.]),
+                              intensities=np.array([0.4, 0.2, 0.1]),
+                              metadata={"precursor_mz": 105.0,
+                                        "testID": "two"})
+
+        # Use factory to construct a similarity function
+        modified_cosine = ModifiedCosine(tolerance=0.2)
+        spectrums = [spectrum_1, spectrum_2]
+        scores = calculate_scores(spectrums, spectrums, modified_cosine)
+        ms_network = SimilarityNetwork(identifier="testID")
+        ms_network.create_network(scores)
+
+        print("Nodes:", ms_network.graph.nodes())
+
+    Should output
+
+    .. testoutput::
+
+        Nodes: ['one', 'two']
+
+    """
+    def __init__(self, identifier: str = "spectrumid",
+                 top_n: int = 20,
+                 max_links: int = 10,
+                 score_cutoff: float = 0.7,
+                 link_method: str = 'single'):
+        """
+        Parameters
+        ----------
+        identifier
+            Unique intentifier for each spectrum in scores. Will also be used for
+            node names.
+        top_n
+            Consider edge between spectrumA and spectrumB if score falls into
+            top_n for spectrumA or spectrumB (link_method="single"), or into
+            top_n for spectrumA and spectrumB (link_method="mutual"). From those
+            potential links, only max_links will be kept, so top_n must be >= max_links.
+        max_links
+            Maximum number of links to add per node. Default = 10.
+            Due to incoming links, total number of links per node can be higher.
```

Comment:
If incoming links can go over the max_links then the order in which scores are filtered matters. Can you say something about order in which spectrums are traversed? 
```suggestion
            Due to incoming links, total number of links per node can be higher. The links are populated by looping over the query spectrums.
```